src/software/robot_diagnostics/widgets/robot_selection.cpp

```diff
@@ -0,0 +1,45 @@
+#include "software/robot_diagnostics/widgets/robot_selection.h"
+
+#include <g3log/g3log.hpp>
+
+void setupRobotSelection(Ui::AutoGeneratedMainWidget *widget,
+                         unsigned int &robot_selection)
+{
+    // clang-format off
+    widget->comboBox_robot_selection->addItems({
+        "1",
+        "2",
+        "3",
+        "4",
+        "5",
+        "6",
+        "7",
+        "8",
+        "9",
+        "10",
+        "11",
+        "12",
+        "13",
+        "14",
+        "15",
+        "16"});
```

Comment:
Does QT not have a specific field for restricted numeric inputs? No worries if not, just want to double check.

Reply:
yes but it's called a spinBox and does not have a nice drop down

Reply:
Hmm, I don't feel super strongly either way. @MathewMacDougall - your thoughts?

Reply:
This is what a QSpinBox looks like: https://doc.qt.io/qt-5/qtwidgets-widgets-spinboxes-example.html I do prefer to have a drop down so people can directly select a number rather than repeatedly incrementing the value.

Reply:
I like the dropdown more than the spinbox. My suggestion would be to add a constant for the `MAX_SSL_ROBOT_ID` (which is 16 like you already have), and then you can populate the widget dynamically in a loop (qt has `QString::number(..)` to create strings from numbers.

Also the list should start at 0 :eyes:

Reply:
done